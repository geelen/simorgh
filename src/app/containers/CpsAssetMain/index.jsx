import React from 'react';
import { string, shape, object, arrayOf } from 'prop-types';
import path from 'ramda/src/path';
import ATIAnalytics from '../ATIAnalytics';
import MetadataContainer from '../Metadata';
import { Grid, GridItemConstrainedMedium } from '../../lib/styledGrid';
import CpsAssetBlocks from '../CpsAssetBlocks';

const CpsAssetMain = ({ pageData }) => {
  const blocks = path(['content', 'blocks'], pageData);
  const promo = path(['promo'], pageData);
  const metadata = path(['metadata'], pageData);

  return (
    <>
      <ATIAnalytics data={pageData} />
      <MetadataContainer metadata={metadata} promo={promo} />
      <Grid as="main" role="main">
        <GridItemConstrainedMedium>
          <CpsAssetBlocks blocks={blocks} />
        </GridItemConstrainedMedium>
      </Grid>
    </>
  );
};

CpsAssetMain.propTypes = {
  pageData: shape({
    metadata: shape({
      id: string,
      tags: object,
    }),
    promo: shape({
      subtype: string,
      name: string,
    }),
    content: shape({
      blocks: arrayOf(
        shape({
          uuid: string,
          id: string,
          externalId: string,
          text: string,
          type: string,
        }),
      ),
    }),
  }).isRequired,
};

export default CpsAssetMain;
