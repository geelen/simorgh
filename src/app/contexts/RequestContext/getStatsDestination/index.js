/* Returns stats destnation for ATI based on origin, service and env
   see table on this issue https://github.com/bbc/simorgh/issues/2995
*/
const getStatsDestination = ({ isUK = true, env = 'test', service }) => {
  let destination = '';
  switch (service) {
    case 'news':
      // checks if "news" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_PS' : 'NEWS_GNL';
      break;
    case 'cymrufyw':
    case 'naidheachdan':
      // checks if "cymrufyw" or "naidheachdan" service is in the UK or not and set apprioprate destination
      destination = isUK !== false ? 'NEWS_LANGUAGES_PS' : 'NEWS_LANGUAGES_GNL';
      break;
    case 'japanese':
      destination = 'NEWS_LANGUAGES_GNL';
      break;
    case 'scotland':
      destination = 'PLACEHOLDER';
      break;
    case 'archive':
      destination = 'BBC_ARCHIVE_PS';
      break;
    default:
      destination = 'WS_NEWS_LANGUAGES';
  }
  return env === 'live' ? destination : `${destination}_TEST`;
};

export default getStatsDestination;
