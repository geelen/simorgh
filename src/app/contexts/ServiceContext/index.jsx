import React from 'react';
import { node, string } from 'prop-types';
import services from '#lib/config/services/loadableConfig';
import { getVariant } from '../../lib/utilities/variantHandler';
import variantPropType from '../../models/propTypes/variants';

export const ServiceContext = React.createContext({});

export const ServiceContextProvider = ({ children, service, variant }) => {
  const LoadableContextProvider = services[service];

  if (!LoadableContextProvider) {
    return null;
  }

  return (
    <LoadableContextProvider
      Context={ServiceContext}
      dataKey={getVariant({ service, variant })}
    >
      {children}
    </LoadableContextProvider>
  );
};

ServiceContextProvider.propTypes = {
  children: node.isRequired,
  service: string,
  variant: variantPropType,
};

ServiceContextProvider.defaultProps = {
  service: 'default',
  variant: 'default',
};
