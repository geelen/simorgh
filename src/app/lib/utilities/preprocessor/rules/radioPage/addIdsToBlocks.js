import pipe from 'ramda/src/pipe';
import path from 'ramda/src/path';
import map from 'ramda/src/map';
import mergeDeepLeft from 'ramda/src/mergeDeepLeft';
import uuid from 'uuid';

const addIdToBlock = block => ({ ...block, uuid: uuid() });

const getBlocks = path(['content', 'blocks']);

const mapBlocks = pipe(getBlocks, map(addIdToBlock));

export default jsonRaw =>
  mergeDeepLeft(
    {
      content: { blocks: mapBlocks(jsonRaw) },
    },
    jsonRaw,
  );
