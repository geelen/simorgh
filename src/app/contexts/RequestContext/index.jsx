import React from 'react';
import { bool, node, oneOf, string, number } from 'prop-types';
import getStatsDestination from './getStatsDestination';
import getStatsPageIdentifier from './getStatsPageIdentifier';
import getOriginContext from './getOriginContext';
import getEnv from './getEnv';
import getMetaUrls from './getMetaUrls';
import variantPropType from '../../models/propTypes/variants';

export const RequestContext = React.createContext({});

export const RequestContextProvider = ({
  children,
  bbcOrigin,
  id,
  isAmp,
  pageType,
  service,
  statusCode,
  previousPath,
  pathname,
  variant,
}) => {
  const { isUK, origin } = getOriginContext(bbcOrigin);
  const env = getEnv(origin);
  const platform = isAmp ? 'amp' : 'canonical';
  const statsDestination = getStatsDestination({
    isUK,
    env,
    service,
  });
  const statsPageIdentifier = getStatsPageIdentifier({
    pageType,
    service,
    id,
  });

  const value = {
    env,
    id,
    isUK,
    origin,
    pageType,
    isAmp,
    platform,
    statsDestination,
    statsPageIdentifier,
    statusCode,
    previousPath,
    variant,
    ...getMetaUrls(origin, pathname),
  };

  return (
    <RequestContext.Provider value={value}>{children}</RequestContext.Provider>
  );
};

RequestContextProvider.propTypes = {
  bbcOrigin: string,
  children: node.isRequired,
  id: string,
  isAmp: bool.isRequired,
  pageType: oneOf([
    'article',
    'frontPage',
    'media',
    'error',
    'MAP',
    'FIX',
    'STY',
    'PGL',
  ]).isRequired,
  service: string.isRequired,
  statusCode: number,
  pathname: string.isRequired,
  previousPath: string,
  variant: variantPropType,
};

RequestContextProvider.defaultProps = {
  bbcOrigin: null,
  id: null,
  statusCode: null,
  previousPath: null,
  variant: null,
};
