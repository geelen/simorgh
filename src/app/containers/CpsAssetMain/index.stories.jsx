import React from 'react';
import { storiesOf } from '@storybook/react';
import { inputProvider } from '@bbc/psammead-storybook-helpers';
import { withKnobs } from '@storybook/addon-knobs';
import { ServiceContextProvider } from '../../contexts/ServiceContext';
import { ToggleContextProvider } from '../../contexts/ToggleContext';
import { RequestContextProvider } from '../../contexts/RequestContext';

import CpsAssetMain from '.';
import indonesia from '../../../../data/indonesia/bbc_indonesian_radio/liveradio.json';
import korean from '../../../../data/korean/bbc_korean_radio/liveradio.json';
import tigrinya from '../../../../data/tigrinya/bbc_tigrinya_radio/liveradio.json';
import afaanoromoo from '../../../../data/afaanoromoo/bbc_afaanoromoo_radio/liveradio.json';
import amharic from '../../../../data/amharic/bbc_amharic_radio/liveradio.json';

// Not all services have fixtures for article data yet
// the service selector will be constrained to services that have article fixtures:
const liveRadioFixtures = {
  indonesia,
  korean,
  tigrinya,
  afaanoromoo,
  amharic,
};

const validServices = Object.keys(liveRadioFixtures);

const matchFixtures = service => ({
  params: {
    mediaId: 'liveradio',
    serviceId: {
      indonesia: 'bbc_indonesian_radio',
      korean: 'bbc_korean_radio',
      tigrinya: 'bbc_tigrinya_radio',
      afaanoromoo: 'bbc_afaanoromoo_radio',
      amharic: 'bbc_amharic_radio',
    }[service],
  },
});

storiesOf('Containers|Cps Asset', module)
  .addDecorator(withKnobs)
  .add(
    'default',
    inputProvider({
      // eslint-disable-next-line react/prop-types
      componentFunction: ({ service }) => {
        return (
          <ToggleContextProvider>
            <ServiceContextProvider service={service}>
              <RequestContextProvider
                isAmp={false}
                pageType="media"
                service={service}
              >
                <CpsAssetMain
                  pageData={liveRadioFixtures[service]}
                  match={matchFixtures(service)}
                  service={service}
                />
              </RequestContextProvider>
            </ServiceContextProvider>
          </ToggleContextProvider>
        );
      },
      services: validServices,
      options: { defaultService: 'indonesia' },
    }),
  );
