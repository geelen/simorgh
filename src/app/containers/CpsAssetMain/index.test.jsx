import React from 'react';
import { latin } from '@bbc/gel-foundations/scripts';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { ServiceContext } from '../../contexts/ServiceContext';
import { RequestContext } from '../../contexts/RequestContext';
import CpsAssetMain from '.';
import amharicPageData from '../../../../data/amharic/bbc_amharic_radio/liveradio';
import addIdsToBlocks from '../../routes/getInitialData/mediapage/addIdsToBlocks';

jest.mock('../Metadata', () => () => <div id="metadata" />);

jest.mock('../CpsAssetBlocks', () => props => <div id="blocks" {...props} />);

const pageData = addIdsToBlocks(amharicPageData);

describe('Cps Asset Main', () => {
  shouldMatchSnapshot(
    'should match snapshot',
    <ServiceContext.Provider value={{ script: latin }}>
      <RequestContext.Provider
        value={{ platform: 'canonical', pageType: 'media' }}
      >
        <CpsAssetMain service="amharic" pageData={pageData} />
      </RequestContext.Provider>
    </ServiceContext.Provider>,
  );
});
