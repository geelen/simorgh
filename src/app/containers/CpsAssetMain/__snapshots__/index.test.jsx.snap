// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media Page Main should match snapshot 1`] = `
.c0 {
  margin: 0 auto;
  padding-bottom: 2rem;
}

.c1 {
  grid-column: 1 / span 6;
}

@media (max-width:37.4375rem) {
  .c0 {
    grid-column-gap: 0.5rem;
  }
}

@media (min-width:37.5rem) {
  .c0 {
    grid-column-gap: 1rem;
  }
}

@media (max-width:62.9375rem) {
  .c0 {
    grid-template-columns: repeat(6,1fr);
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(8,minmax(0,6.75rem)) 1fr;
    max-width: 45.5rem;
  }
}

@media (min-width:80rem) {
  .c0 {
    grid-template-columns: 1fr repeat(20,minmax(0,2.95rem)) 1fr;
    max-width: 46.4rem;
  }
}

@supports (display:grid) {
  .c0 {
    display: grid;
    max-width: initial;
    margin: initial;
  }
}

@media (max-width:25rem) {
  .c1 {
    padding: 0 0.5rem;
  }
}

@media (min-width:25rem) and (max-width:62.9375rem) {
  .c1 {
    padding: 0 1rem;
  }
}

@media (min-width:37.5rem) {
  .c1 {
    grid-column: 1 / span 5;
    max-width: 83.33%;
  }
}

@media (min-width:63rem) and (max-width:80rem) {
  .c1 {
    grid-column: 3 / span 5;
    max-width: 37.75rem;
  }
}

@media (min-width:80rem) {
  .c1 {
    grid-column: 6 / span 10;
    max-width: 38.5rem;
  }
}

@supports (display:grid) {
  .c1 {
    max-width: initial;
  }
}

<div>
  <noscript />
  <div
    id="metadata"
  />
  <main
    class="c0"
    role="main"
  >
    <div
      class="c1"
    >
      <div
        blocks="[object Object],[object Object],[object Object]"
        id="blocks"
      />
    </div>
  </main>
</div>
`;
